module/mfrc522_spi.c

```diff
@@ -18,17 +20,18 @@ u8 mfrc522_register_read(struct spi_device *client, u8 reg)
 		address_byte_build(MFRC522_SPI_READ, reg);
 	u8 value;
 
-	spi_write(client, &reg_read, 1);
-	spi_read(client, &value, 1);
-
+	spi_write_then_read(client, &reg_read, 1, &value, 1);
 	return value;
 }
 
 void mfrc522_register_write(struct spi_device *client, u8 reg, u8 value)
 {
 	struct address_byte reg_write =
 		address_byte_build(MFRC522_SPI_WRITE, reg);
+	u8 data[2] = { 0, value };
+
+	// We cannot directly put reg_write in data[0] at init time
+	memcpy(&data[0], &reg_write, sizeof(u8));
```

Comment:
```suggestion
	data[0] = reg_write;
```
?

Reply:
Not working:
```
mfrc522_spi.c:30:2: error: aggregate value used where an integer was expected
```

Reply:
Oh my bad I had completely forgotten that we had a struct address byte. I thought we were still dealing with u8. Perfect then! Good job and thanks